src/std/misc/string-test.ss

```diff
@@ -3,12 +3,18 @@
 (import
   :std/misc/string :std/srfi/13
   :std/test :gerbil/gambit/exceptions
-  :std/pregexp)
+  :std/pregexp :std/misc/repr :std/sugar)
 
 (def (error-with-message? message)
   (lambda (e)
     (and (error-exception? e) (equal? (error-exception-message e) message))))
 
+(defstruct point (x y))
+(defmethod {:pr point}
+  (lambda (self port options)
+    (for-each (cut display <> port)
+              ["(point " (point-x self) " " (point-y self) ")"])))
```

Comment:
you can just `fprintf` here, much simpler code.

Reply:
I copied the example from [:std/misc/repr](https://cons.io/reference/misc.html#representable).

Reply:
that should be fixed too then :) of course it's ok if you don't feel like adding it in this pr.

Reply:
I've anyway planed to fix/write some documentation.